docs/features/non-source-outputs.md

```diff
@@ -0,0 +1,141 @@
+
+By default the supplied `hintName` will be used as the filename. We already have conditions in place for source texts to prevent hint names from being invalid filenames, and we will reuse this logic for artifact names. If two generators attempt to write a file with the same name, the generator name, including namespace, will be prepended to the name of the files to ensure there are no conflicts.
+
+### Customizing output locations
+
+While it may be acceptable for the majority of cases to simply place the artifacts in the bin directory, there are known cases (such as the Javascript interop case mentioned at the beginning) where it would be beneficial to place generated files in a more specific location (such as `wwwroot`). This proposal provides a mechanism to override the location of artifacts in two ways: on a per-generator basis, and on an individual artifact basis. Both of these mechanisms are implemented via `.globalconfig` options, and exposed with `MSBuild` properties to enable easy configuration by both Generator authors and consumers.
+
+The generator driver will recognize the following `.globalconfig` options:
+
+- `compiler_artifact.<generatorname>.outputdir = <dir>`
+- `compiler_artifact.<artifactname>.outputdir = <dir>`
+
+When the second part of the key matches the name of a loaded generator, any artifacts produced by that generator will be placed in the specified `<dir>` instead of the default directory. If the key fails to match a generator, individual artifacts are checked: if the hint name of an artifact matches it will be placed in the specified `<dir>`.
+
+While these properties can be set manually in a `.globalconfig` it's expected that they will instead be passed from `MSBuild` as project items. The compiler already has a mechanism via `.MSBuildGeneratedEditorConfig` that allows the mapping of `MSBuild` information into `.globalconfig` format, and this proposal expands on that to support the compiler artifact mapping. The mapping will recognize the `<CompilerArtifactMapping` item, and will map the `include` value into the key name and the `Location` attribute as the key value (`<dir>`)
+
+```xml
+<ItemGroup>
+    <!-- Map all of the outputs of the razor source generator -->
+    <CompilerArtifactMapping Include="Microsoft.CodeAnalysis.Razor.SourceGenerators.SourceGenerator" Location="$(OutDir)\Razor" />
+
+    <!-- Map an individual artifact -->
+    <CompilerArtifactMapping Include="ArtifactName" Location="c:\dir" />
+<ItemGroup>
+```
+
+Would produce the following in the generated editor config:
+
+```ini
+compiler_artifact.Microsoft.CodeAnalysis.Razor.SourceGenerators.SourceGenerator.outputdir = bin\net8.0\Razor
+compiler_artifact.ArtifactName.outputdir = c:\dir
```

Comment:
why compiler_artifact, but not output_dir?  the difference in naming is tweaking me.